enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts

```diff
@@ -36,8 +49,8 @@ export function useLoadQuestion({
 
   // Keep track of the latest question and query results.
   // They can be updated from the below actions.
-  const [result, setQuestionResult] = useState<SdkQuestionResult>({});
-  const { question, queryResults } = result;
+  const [questionState, setQuestionState] = useReducer(questionReducer, {});
```

Comment:
I think when the `cardId` changes, the reference to the `loadQuestion` method changes, and we have a `useEffect` to call `loadQuestion` so `setQuestionState` should be called there to reset the state. It's very non-intuitive thought, so I'll document this a bit. I agree it would be better to be more explicit and have `cardId` in the dependency array.

```ts
  const {
    question,
    originalQuestion,

    queryResults,

    isQuestionLoading,
    isQueryRunning,

    runQuestion,
    loadQuestion,
    updateQuestion,
    navigateToNewCard,
  } = useLoadQuestion({
    cardId,
    options,
    deserializedCard,
  });


  useEffect(() => {
    loadQuestion();
  }, [loadQuestion]);
```